### PR TITLE
Allow querying by custom properties in `@listing` endpoint.

### PR DESCRIPTION
**Pull request by deiferni**
We add support to query the `@listing` endpoint with custom properties/dynamic fields. Available custom columns by listing are provided globally in a `@listing-custom-fields` endpoint.

The custom fields are provided in a global endpoint on the plone site, so that the UI can request these mostly static information once, and then retrieve it from from global application state rather than querying the available fields every time before a listing is requested from the backend.

The `@listing-custom-fields` endpoint returns all available fields based on currently registered property sheets. It takes care of associating property sheets to available listing names and merges fields from different property sheets with the same name and type while building the list.

The `@listing` endpoint now allows all those additional fields to be retrieved via the `columns` argument.

I have moved the responsibility/knowledge about what fields can be indexed in solr to the property sheet schema definition class and provided a distinct field class for solr fields with `SolrDynamicField`.

I have discovered some assumptions built into https://github.com/4teamwork/opengever.core/pull/6959 which are not yet completely validated upon input. The changes here also silently rely on certain states not being introduced into the system. If you do (which you currenlty can) behavior is undefined
- make sure default slot and type-based slot property names don't overlap
- only allow string type values in choices

_I will amend validation in a separate PR._

Jira: https://4teamwork.atlassian.net/browse/CA-1494

## Checklist

_Everything has to be done/checked. Checked but not present means the author deemed it unnecessary._

- [x] Changelog entry
- [x] Link to issue (Jira or GitHub) and backlink in issue (Jira)

_Only applicable should be left and checked._

- API change:
  - [x] Documentation is updated
  - [x] API Changelog entry (see [guide](https://4teamwork.atlassian.net/wiki/spaces/4TEAM/pages/451248812/API+Changelog+Guidelines))

### CODE REVIEW

File: opengever/api/listing_custom_fields.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+
+class ListingCustomFieldsGet(Service):
+    """API Endpoint which returns custom fields available for listings.
+
+    It returns a nested data structure with custom fields for each supported
+    listing, if available.
+    Custom fields are provided as follows:
+    - Custom field source are propety sheets registerd for a type associated
```
Comment (njohner) [inline opengever/api/listing_custom_fields.py:19]: propety -> property

File: docs/public/dev-manual/api/listings.rst [hunk 2]
```diff
@@ -243,6 +245,54 @@ siehe Tabelle:
...
     +--------------------------+----------+---------+--------------+--------------------+---------+---------+---------+----------+-----------------+------------------+-----------------+----------+------------------+----------------+
 
+.. _listing-property_sheets:
+
+Benutzerdefinierte Felder:
+--------------------------
+
+Falls :ref:`benutzerdefinierte Felder <propertysheets>` definiert sind, stehen
+mit dem Endpoint ``@listing-custom-fields`` weitere, dynamische Felder zur
+Verfügung. Der Endpoint kann z.B. dafür benutzt werden für ein Filtermenü für
```
Comment (njohner) [inline docs/public/dev-manual/api/listings.rst:255]: für ein Filtermenü für -> für ein Filtermenü oder für

File: docs/public/dev-manual/api/listings.rst [hunk 3]
```diff
@@ -243,6 +245,54 @@ siehe Tabelle:
...
+sichtbare Spalten darzustellen. Er leifert `Titel`, `Typ` und `Name` zurück.
```
Comment (njohner) [inline docs/public/dev-manual/api/listings.rst:256]: `Titel`, `Typ` und `Name` zurück. -> `titel`, `type` und `name` zurück.

Comment (njohner) [reply docs/public/dev-manual/api/listings.rst:256]: and shouldn't you use double ticks here?

**Review by njohner — APPROVED** [review]
LGTM 🎉

File: docs/public/dev-manual/api/listings.rst (hunk 3 above)
Comment (deiferni) [reply docs/public/dev-manual/api/listings.rst:256]: I was attempting to do it descriptive, without the identifiers. But it seems confusing, i have changed it as you have suggested.

**Pull request merged**